src/components/input/NumericInput.js

```diff
@@ -89,35 +96,80 @@ class NumericInput extends PureComponent {
 
   isMinReached = () => this.state.value <= this.props.min;
 
-  getSuffixWithSpinner = () => [
-    ...this.props.suffix,
-    /* eslint-disable-next-line react/jsx-key */
-    <SpinnerControls
-      inverse={this.props.inverse}
-      spinnerUpProps={{
-        onClick: this.handleIncreaseValue,
-        disabled: this.isMaxReached(),
-      }}
-      spinnerDownProps={{
-        onClick: this.handleDecreaseValue,
-        disabled: this.isMinReached(),
-      }}
-    />,
-  ];
+  getConnectedRight = () => {
+    const { connectedRight, size, stepper } = this.props;
+
+    if (stepper === 'connected') {
+      return (
+        <Button
+          disabled={this.isMaxReached()}
+          icon={<IconAddSmallOutline />}
+          onClick={this.handleIncreaseValue}
+          size={size}
+        />
+      );
+    }
+
+    return connectedRight;
```

Comment:
Q: not entirely sure what this returns? wouldn't `null` suffice?

Reply:
A: `connectedRight` and `connectedLeft` are props that can contain f.e. a `Button`. But, when the stepper controls are rendered left and right, we don't want any extra connected component being rendered.

Reply:
Simply speaking, if we have a `<NumericInput stepper="connected" connectedRight={<Button />}>` the `connectedRight` would be overwritten.

Reply:
@driesd @rathesDot thanks for explaining, missed that context. Still feels like an odd architecture though.